src/lib/sql/sql_translator.cpp

```diff
@@ -934,12 +936,33 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_predicate_expression(
 
       switch (logical_expression->logical_operator) {
         case LogicalOperator::And: {
-          current_node = _translate_predicate_expression(logical_expression->right_operand(), current_node);
-          return _translate_predicate_expression(logical_expression->left_operand(), current_node);
+          current_node =
+              _translate_predicate_expression(logical_expression->right_operand(), current_node, or_translation_mode);
+          return _translate_predicate_expression(logical_expression->left_operand(), current_node, or_translation_mode);
         }
         case LogicalOperator::Or: {
-          current_node = _add_expressions_if_unavailable(current_node, {expression});
-          return PredicateNode::make(not_equals_(expression, 0), current_node);
+          /**
+           * We have two ways to process OR predicates: Via a Projection, which also supports more complex expressions
+           * like "a + 5 > 3 OR b + c > d" and is likely faster - and via UnionPositions which has the advantage of
+           * outputting ReferenceColumns.
```

Comment:
Does this mean that `SELECT * FROM t WHERE a + 5 > 3 OR b + c > d` works but `DELETE FROM t WHERE a + 5 > 3 OR b + c > d` not?

Reply:
precisely. The second one will still yield the error that Dimitri reported

Reply:
Mmmh, this looks like a really dirty hack to fix something that is not a show-stopper.

`SELECT ... FROM t WHERE expr` should behave exactly like `DELETE FROM t WHERE expr`, the only difference that the second removes the found tuples.

Reply:
ExpressionEvaluator::evaluate_expression_to_pos_list would support that, though.

Reply:
> Mmmh, this looks like a really dirty hack to fix something that is not a show-stopper

I was under the impression that it is a show stopper for Dimitri.
Find it less dirty than the alternative, which would be using UnionPositions in places where it would only limit performance just so that `SELECT ... FROM t WHERE expr` and `DELETE FROM t WHERE expr` can produce the same LQP

Reply:
I understand that it is only used in tests that modify the table so that log entries are generated. These would work without the OR clause, too.

Maybe this is something that is easier to discuss in person.

Reply:
Well, I can write tests without OR statements, so it's not a problem that has to be fixed asap.